eZ/Publish/API/Repository/Values/User/RoleCreateStruct.php

```diff
@@ -18,6 +18,13 @@
 abstract class RoleCreateStruct extends ValueObject
 {
     /**
+     * ID of a role.
+     *
+     * @var int
+     */
+    public $id;
```

Comment:
See my comment about `createRole()` / `createRoleDraft()`.
We should not have an ID in a create/update struct. Problem comes with how role / role draft creation is made with your current change.